core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -482,6 +506,7 @@ public TableMetadata updateSchema(Schema newSchema, int newLastColumnId) {
     // rebuild all of the partition specs and sort orders for the new current schema
     List<PartitionSpec> updatedSpecs = Lists.transform(specs, spec -> updateSpecSchema(newSchema, spec));
     List<SortOrder> updatedSortOrders = Lists.transform(sortOrders, order -> updateSortOrderSchema(newSchema, order));
+    RowKey updatedRowKey = updateRowKeySchema(newSchema, rowKey);
```

Comment:
should we fail if schema update drops column defined in the row key? probably want to add test for that too

Reply:
That's a good point, I was thinking about having another PR for blocking column drop for column in row key after this one, do you think it's better to have it here?

Reply:
I think it would be a relatively small code change in this class and some testing which might be easy to include in this same PR, unless you are thinking about something different?

Reply:
I was thinking about the following: 1 PR for the update related changes including update row key API, implementation, and this one; 1PR for work in Spark for SQL extension; 1PR in Flink for primary key clause (if no one did it yet)

Reply:
@jackye1995  your plan looks great !